janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java

```diff
@@ -179,93 +179,101 @@ public CQLStoreManager(final Configuration configuration) throws BackendExceptio
         this.keyspace = determineKeyspaceName(configuration);
         this.openStores = new ConcurrentHashMap<>();
         this.session = initializeSession();
-        initializeJmxMetrics();
-        initializeKeyspace();
 
-        int batchSize = configuration.get(BATCH_STATEMENT_SIZE);
-        boolean atomicBatch = configuration.get(ATOMIC_BATCH_MUTATE);
+        try{
 
-        if(configuration.get(EXECUTOR_SERVICE_ENABLED)){
-            this.executorService = buildExecutorService(configuration);
-            if(atomicBatch){
-                executeManyFunction = new CQLExecutorServiceMutateManyLoggedFunction(times,
-                    assignTimestamp, openStores, session, executorService, this::sleepAfterWrite);
-            } else {
-                executeManyFunction = new CQLExecutorServiceMutateManyUnloggedFunction(batchSize,
-                    session, openStores, times, executorService, assignTimestamp, this::sleepAfterWrite);
-            }
-        } else {
-            this.executorService = null;
-            if(atomicBatch){
-                executeManyFunction = new CQLSimpleMutateManyLoggedFunction(times,
-                    assignTimestamp, openStores, session, this::sleepAfterWrite);
+            initializeJmxMetrics();
```

Comment:
This try block seems quite long. Do you consider moving the logic within try block to a separate method?

Reply:
There are some `final` fields to which we assign value in constructor: `executorService`, `executeManyFunction`, `deployment`, `storeFeatures`. We can assign values to these fields in constructor only, thus we can't simply move this logic to a separate method. I can just refactor the logic to pass maybe builders to the `CQLStoreManager` constructor and than trigger those builders and assign the result values to these final fields. What do you think about it?

Reply:
That sounds great to me! Otherwise it's fine to keep it as it is.